x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/inference/adaptiveallocations/AdaptiveAllocationsScalerService.java

```diff
@@ -415,49 +415,59 @@ private void processDeploymentStats(GetDeploymentStatsAction.Response statsRespo
                 if (newNumberOfAllocations > numberOfAllocations.get(deploymentId)) {
                     lastScaleUpTimesMillis.put(deploymentId, now);
                 }
-                UpdateTrainedModelDeploymentAction.Request updateRequest = new UpdateTrainedModelDeploymentAction.Request(deploymentId);
-                updateRequest.setNumberOfAllocations(newNumberOfAllocations);
-                updateRequest.setIsInternal(true);
-                ClientHelper.executeAsyncWithOrigin(
-                    client,
-                    ClientHelper.ML_ORIGIN,
-                    UpdateTrainedModelDeploymentAction.INSTANCE,
-                    updateRequest,
-                    ActionListener.wrap(updateResponse -> {
-                        logger.info("adaptive allocations scaler: scaled [{}] to [{}] allocations.", deploymentId, newNumberOfAllocations);
-                        threadPool.executor(MachineLearning.UTILITY_THREAD_POOL_NAME)
-                            .execute(
-                                () -> inferenceAuditor.info(
-                                    deploymentId,
-                                    Strings.format(
-                                        "adaptive allocations scaler: scaled [%s] to [%s] allocations.",
-                                        deploymentId,
-                                        newNumberOfAllocations
-                                    )
-                                )
-                            );
-                    }, e -> {
-                        logger.atLevel(Level.WARN)
-                            .withThrowable(e)
-                            .log(
-                                "adaptive allocations scaler: scaling [{}] to [{}] allocations failed.",
-                                deploymentId,
-                                newNumberOfAllocations
-                            );
-                        threadPool.executor(MachineLearning.UTILITY_THREAD_POOL_NAME)
-                            .execute(
-                                () -> inferenceAuditor.warning(
-                                    deploymentId,
-                                    Strings.format(
-                                        "adaptive allocations scaler: scaling [%s] to [%s] allocations failed.",
-                                        deploymentId,
-                                        newNumberOfAllocations
-                                    )
-                                )
-                            );
-                    })
-                );
+                updateNumberOfAllocations(deploymentId, newNumberOfAllocations);
             }
         }
     }
+
+    public boolean maybeStartAllocation(TrainedModelAssignment assignment) {
+        if (assignment.getAdaptiveAllocationsSettings() != null
```

Comment:
To be safe pls protect this with a feature flag check in case there is some weird corner case during scaling where the assignment exists but not on any nodes yet 

```suggestion
        if (ScaleToZeroFeatureFlag.isEnabled() && assignment.getAdaptiveAllocationsSettings() != null
```

Reply:
Thanks, guess that's slightly safer. Committed your change. (Not that it matters that much; the condition to get here should in theory never trigger without scaling to zero enabled.)